deltachat-ffi/src/lib.rs

```diff
@@ -28,7 +28,14 @@ pub unsafe extern "C" fn dc_context_new(
 ) -> *mut dc_context_t {
     setup_panic!();
 
-    let ctx = context::dc_context_new(cb, userdata, os_name);
+    let rust_cb: Option<dc_callback_t>;
+    if cb as libc::uintptr_t != 0 {
+        rust_cb = Some(cb);
+    } else {
+        rust_cb = None;
+    }
```

Comment:
this should be written as 

let rustcb = if { .. } else { .. };

Reply:
you can change the function signature to be Option<callback> and rust ffi does the null pointer to none logic automatically

Reply:
see https://doc.rust-lang.org/nomicon/ffi.html#the-nullable-pointer-optimization

Reply:
Thanks for that link, updated.  I must say the implicit raw-pointer to fn-pointer conversion did confuse me for a bit (one of my earlier attempts was to turn this into a raw pointer) but I guess these abstractions become handy once you're used to them.

I also keep forgetting you can use if as an expression, thanks for the reminder.